cli_tools/osconfig_agent/main.go

```diff
@@ -33,6 +40,58 @@ var (
 
 var dump = &pretty.Config{IncludeUnexported: true}
 
+const (
+	// TODO: make this configurable.
+	interval    = 10 * time.Minute
+	metadataURL = "http://metadata.google.internal/computeMetadata/v1/instance/?recursive=true&alt=json"
+)
+
+type metadataJSON struct {
+	ID   int
+	Zone string
+}
+
+func getResource(r string) (string, error) {
+	if r != "" {
+		return r, nil
+	}
+
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", metadataURL, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Add("Metadata-Flavor", "Google")
+
+	var res *http.Response
+	// Retry forever, increase sleep between retries (up to 5 times) in order
+	// to wait for slow network initialization.
+	var rt time.Duration
+	for i := 1; ; i++ {
+		res, err = client.Do(req)
+		if err == nil {
+			break
+		}
+		if i < 6 {
+			rt = time.Duration(3*i) * time.Second
```

Comment:
minor: you can simplify this a little by doing something like `rt = time.Duration(math.Min(3 * i, MAX_RETRY_DELAY)) * time.Second`. Then you can remove the `if i < 6` check and you don't need to track the state of `rt` between iterations to understand how long the delay is. It will also make it easier to extend in the future to add exponential backoff and jitter.